documentation/projects/proposals/ingestion_server_removal/20240328-implementation_plan_ingestion_server_removal.md

```diff
@@ -0,0 +1,586 @@
+
+### Infrastructure
+
+<!-- Describe any infrastructure that will need to be provisioned or modified. In particular, identify associated potential cost changes. -->
+
+Infrastructure is a large part of this project. We will be adding 8 new EC2
+instances but deprovisioning ten, as described above. There will be significant
+cost benefits to removing the two EC2 instances which are constantly running.
+
+### Tools & packages
+
+<!-- Describe any tools or packages which this work might be dependent on. If multiple options are available, try to list as many as are reasonable with your own recommendation. -->
+
+No new tools or packages are required.
+
+### Other projects or work
+
+<!-- Note any projects this plan is dependent on. -->
+
+- [Deploying Airflow with Ansible](https://github.com/WordPress/openverse-infrastructure/pull/829),
+  part of the
+  [Move Airflow to openverse.org](https://github.com/WordPress/openverse-infrastructure/milestone/6)
+  Project.
+- [Catalog Data Cleaning project](https://github.com/WordPress/openverse/issues/430),
+  which removes the Clean Data steps from the data refresh
+
+## Alternatives
+
+<!-- Describe any alternatives considered and why they were not chosen or recommended. -->
+
+The alternative options of using an ECS approach or performing the reindex
+entirely in Airflow are discussed at length in the [Approach to the Distributed
+Reindex][#approach-to-the-distributed-reindex] section.
+
+## Blockers
+
+<!-- What hard blockers exist that prevent further work on this project? -->
+
+Development can start immediately, but the infrastructure components of this
+project are blocked by completion of the effort to move Airflow to
+openverse.org, specifically
+[this issue](https://github.com/WordPress/openverse-infrastructure/issues/776)
+to run a (legacy) data refresh on the new instance.
+
+## Rollback
+
+<!-- How do we roll back this solution in the event of failure? Are there any steps that can not easily be rolled back? -->
+
+We will not remove the ingestion server or the existing data refresh DAGs until
+the new DAGs have been run successfully against both staging and production. We
+can rollback at any time by simply removing the new instances and DAGs.
+
+## Risks
+
+<!-- What risks are we taking with this solution? Are there risks that once taken can’t be undone?-->
+
+There is minimal risk, as we will be able to test against production data in
+staging before running against production. However the data refresh does
+completely replace the production media tables and Elasticsearch indices, so
+there is always inherent risk that production data could be lost or malformed if
+something goes wrong. We will ensure production backups are available and
+monitor the first production data refreshes closely.
+
+## Prior art
+
+<!-- Include links to documents and resources that you used when coming up with your solution. Credit people who have contributed to the solution that you wish to acknowledge. -->
```

Comment:
Should we remove this section? Or reference the search relevancy sandbox project here?